redbeams/src/main/java/com/sequenceiq/redbeams/flow/redbeams/start/actions/AbstractRedbeamsStartAction.java

```diff
@@ -0,0 +1,93 @@
+package com.sequenceiq.redbeams.flow.redbeams.start.actions;
+
+import com.sequenceiq.cloudbreak.cloud.context.CloudContext;
+import com.sequenceiq.cloudbreak.cloud.model.CloudCredential;
+import com.sequenceiq.cloudbreak.cloud.model.Location;
+import com.sequenceiq.cloudbreak.common.event.Payload;
+import com.sequenceiq.cloudbreak.logger.MDCBuilder;
+import com.sequenceiq.flow.core.AbstractAction;
+import com.sequenceiq.flow.core.FlowParameters;
+import com.sequenceiq.redbeams.converter.cloud.CredentialToCloudCredentialConverter;
+import com.sequenceiq.redbeams.domain.stack.DBStack;
+import com.sequenceiq.redbeams.dto.Credential;
+import com.sequenceiq.redbeams.flow.redbeams.common.RedbeamsFailureEvent;
+import com.sequenceiq.redbeams.flow.redbeams.start.RedbeamsStartContext;
+import com.sequenceiq.redbeams.flow.redbeams.start.RedbeamsStartEvent;
+import com.sequenceiq.redbeams.flow.redbeams.start.RedbeamsStartState;
+import com.sequenceiq.redbeams.service.CredentialService;
+import com.sequenceiq.redbeams.service.stack.DBStackService;
+import org.springframework.statemachine.StateContext;
+
+import javax.annotation.PostConstruct;
+import javax.inject.Inject;
+import java.util.Map;
+import java.util.Optional;
+
+import static com.sequenceiq.cloudbreak.cloud.model.AvailabilityZone.availabilityZone;
+import static com.sequenceiq.cloudbreak.cloud.model.Location.location;
+import static com.sequenceiq.cloudbreak.cloud.model.Region.region;
+
+public abstract class AbstractRedbeamsStartAction<P extends Payload>
```

Comment:
AbstractRedbeamsStartAction and AbstractRedbeamsStopAction are very similar, are almost the same. Also, they are quite similar to AbstractRedbeamsProvisionAction and TerminationAction. I would try to create one common class from them, AbstractRedbeamsAction and use it in all four flows.

If that is too big of a task (~more than an 1-2 hours of work), then I would create a separate refactor jira for it.

Reply:
If you don't mind, I'd more prefer the refactor as a separated task.